addons/website/static/src/client_actions/website_preview/website_preview.js

```diff
@@ -95,22 +95,22 @@ export class WebsitePreview extends Component {
             if (this.props.action.context.params && this.props.action.context.params.with_loader) {
                 this.websiteService.showLoader({ showTips: true });
             }
+        });
 
-            return () => {
-                this.websiteService.currentWebsiteId = null;
-                this.websiteService.websiteRootInstance = undefined;
-                this.websiteService.pageDocument = null;
-            };
-        }, () => [this.props.action.context.params]);
-
-        useEffect(() => {
+        onMounted(() => {
```

Comment:
2 onMounted() in the same class?

Reply:
Yes, nothing prevents it, they will both be played

Reply:
Why would you do that though?

Reply:
Because they do two different things. The first one parses the client action params to populate the context, and the second one blocks the iframe while it is loading. I did that to respect the separation of concerns principle.

Reply:
"What should happen at the start" is a "concern" small enough in this case. You case always separate them by nice inline comments which would be better and less confusing to the external reader.

Reply:
I think we are missing one of the advantages of owl by doing this. Instead of putting a lot of code doing different things into one start function, you can add as many hooks as you need, each responsible for a single task.

If later, its is decided that the iframe should not be blocked while loading, it will be simpler to remove that one hook that does it than going through a bigger function that does everything at the first render of the component. It seems like that particular example will not happen, but it will be the same for other hooks.

Having multiple hooks will make the component more modular as well, the code of the hook can be put in a properly named method (which will help readability - maybe these two hooks could already call two methods `blockIframe` and `parseActionParams`), and it will be easier for other modules to override and change that particular behaviour of the component.

Reply:
> maybe these two hooks could already call two methods blockIframe and parseActionParams

I could arguably understand that the (unique) `onMounted()` method would call 2 sub method if we really want to encapsulate code differently than with inline comments, but 2 `onMounted()` calling each one sub method.. Meh :thinking: 

About the current discussion, I really find it super weird to have multiple `onMounted()`, while you seem to think that it is a owl best practice, I just feel it is weird, if I were to search in the code or debug a code, and I would be looking at the `onMounted()` method, I would never expect to find other ones for the same component/class.
I mean, will we have 32 `onMounted()` for a big component?

I don't want to stick on my position if I am wrong and if it is indeed an owl best practice, so maybe @aab-odoo could give us his opinion here.

Reply:
> If later, its is decided that the iframe should not be blocked while loading, it will be simpler to remove that one hook that does it than going through a bigger function

If you comment your code correctly, it won't be easier, it will be exactly the same.

> the code of the hook can be put in a properly named method (which will help readability [...]

It may be an unpopular opinion but I think this is very wrong (as I repeat it in all my reviews). Splitting into sub-functions does not help readability if you comment your code properly (especially if you would name your method `blockIframe` which may be clear to you but is not to anyone else). It surely does help modularity (and thus possibility of extensions) but not readability. More often than not, splitting into sub-functions code that is called only once actually hurts readability IMO: you force the reader to jump through the code for no reason (and often the dev did not comment after its split judging his method name was enough so you are even more lost).

I guess the only real argument in naming two sub-functions in this case would be to have that method name during debugging / traceback... but I think it's not justified enough to *hurt* (IMO) readability.

Reply:
BTW there is only one other occurrence of someone doing this (except one other in OWL codebase):
![image](https://user-images.githubusercontent.com/10338094/185893684-30bfe750-40ca-4082-b916-6752e424bf88.png)

And I cannot think that creating a `onMounted` method called by the `onMounted` hook can be a good pratice (although there are many occurrences of that :grimacing:

Reply:
Thank you for the explanations, I understand better how it can be preferable to keep the onMounted code in one hook.

I pushed a new version of the commit:
- The initial useEffect, parsing the client action params, and called at each changes of this prop, is kept. 
I removed it for the use of an onMounted at first, as for now, this prop does not change throughout the component's lifecycle. 
In the end, I think it is clearer to keep the useEffect, with its dependency, even if it will be called as often as the onMounted for now: on change of the params, the context must be populated with values parsed from them.
(And maybe later it will be possible possible to change an action's behaviour through this prop, at least, it will be ready).
- To fix the bug, the cleanup function that was returned in the useEffect (and bypassed by the an early return when an action was replayed) is moved to an onWillUnmount hook
-  The only remaining onMounted hook blocks the iframe until its first load event.

What do you think about it ?

Out of curiosity I'm still interested in knowing the framework-js team recommendations about using multiple onMounted on one component, and on defining class methods called from these hooks. Thanks in advance :)

Reply:
> Out of curiosity I'm still interested in knowing the framework-js team recommendations about using multiple onMounted on one component, and on defining class methods called from these hooks. Thanks in advance :)

I don't think @aab-odoo still loves me, I already pinged him [a few months ago](https://github.com/odoo/odoo/pull/90360#issuecomment-1117064645) on a joke and he savagely just ignored me :crying_cat_face: 

But a few years ago, we sent an undercover website member in the JS framework team for a 5 years long secret mission.
He will definitely reply and gives his opinion within the hour, he is a quick man, but also a wise man. Exactly what we are looking for.
/summon @sdegueldre

Reply:
Loving you is hard work @rdeodoo and not every father love his son :grimacing: cc @aab-odoo

Reply:
Yep, my Odoo daddy is just using me for easy baby-foot victories here and there at this point.
https://github.com/odoo/odoo/pull/97924#event-7169469254 :roll_eyes:

Reply:
There's no problem with calling multiple times the same (lifecycle) hook. We sometimes do that to properly separate the logic.

Reply:
Then let's agree it's my choice to say it's a readability problem in this case @younn-o :smile:

Reply:
Changed according to https://github.com/odoo/odoo/pull/98352#discussion_r952568204
It is ready for review